### ISSUE TITLE
SC2236/SC2237-equivalent to binary operators: `=`, `!=`, `-eq`, `-ne`, `-gt`, `-ge`, `-lt`, and `-le`

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3164]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
[ ! "$USER" = me ]
! [ "$USER" != me ]
[ ! $$ -eq 0 ]
! [ $$ -gt 0 ]
```

#### Here's what shellcheck currently says:
nothing

#### Here's what I wanted or expected to see:

```
[ ! "$USER" = me ]
  ^-- SCXXXX (style): Use a != b instead of ! a = b
! [ "$USER" != me ]
^-- SCXXXX (style): Use [ a = b ] instead of ! [ a != b ]
[ ! $$ -eq 0 ]
  ^-- SCXXXX (style): Use a -ne b instead of ! a -eq b
! [ $$ -gt 0 ]
^-- SCXXXX (style): Use [ a -le b ] instead of ! [ a -gt b ]
```

similarly to [SC2236](https://www.shellcheck.net/wiki/SC2236) and [SC2237](https://www.shellcheck.net/wiki/SC2237) (double negation).
